### PR TITLE
iadd isub __contains__ improvement

### PR DESCRIPTION
**Pull request by idk3**

### CODE REVIEW

**Comment by idk3** [comment]
I need to think more about why this is failing..

File: src/openfermion/ops/_fermion_operator.py [hunk 1]
```diff
@@ -572,14 +576,10 @@ def __iadd__(self, addend):
...
-                    if abs(addend.terms[term] +
-                           self.terms[term]) < EQ_TOLERANCE:
-                        del self.terms[term]
-                    else:
-                        self.terms[term] += addend.terms[term]
-                else:
-                    self.terms[term] = addend.terms[term]
+                self.terms[term] = (self.terms.get(term, 0.0) +
+                                    addend.terms[term])
+                # if abs(self.terms[term]) < EQ_TOLERANCE:
```
Comment (babbush) [inline src/openfermion/ops/_fermion_operator.py]: What's going on here, you are commenting out lines?

File: src/openfermion/ops/_fermion_operator.py [hunk 2]
```diff
@@ -610,14 +610,10 @@ def __isub__(self, subtrahend):
...
-                    if abs(self.terms[term] -
-                           subtrahend.terms[term]) < EQ_TOLERANCE:
-                        del self.terms[term]
-                    else:
-                        self.terms[term] -= subtrahend.terms[term]
-                else:
-                    self.terms[term] = -subtrahend.terms[term]
+                self.terms[term] = (self.terms.get(term, 0.0) -
+                                    subtrahend.terms[term])
+                # if abs(self.terms[term]) < EQ_TOLERANCE:
```
Comment (babbush) [inline src/openfermion/ops/_fermion_operator.py]: ?

File: src/openfermion/ops/_fermion_operator.py [hunk 3]
```diff
@@ -262,7 +262,7 @@ def __init__(self, term=None, coefficient=1.):
             with a scalar as calls an out-of-place multiplication.
 
         Args:
-            term (tuple of tuples, a string, or optional):
+            term (tuple or list of tuples, a string, or optional):
```
Comment (babbush) [inline src/openfermion/ops/_fermion_operator.py]: Does this PR make sense without your other PR being merged? What do you want to do about that one?

File: src/openfermion/ops/_fermion_operator.py (hunk 1 above)
Comment (idk3) [reply src/openfermion/ops/_fermion_operator.py]: I was trying to figure out the cause of the errors with those lines included - this suggests there's something sketchy about `del self.terms[term]` causing the test failures in JelliumTest. The failures also seem to only happen in Python 2.7 which is pretty weird.

Comment (babbush) [reply src/openfermion/ops/_fermion_operator.py]: Are you going to finish this PR or should I close it?

Comment (idk3) [reply src/openfermion/ops/_fermion_operator.py]: I'll take a look more later this week or next - at this point it looks to be debugging something weird in the jellium module which I'm not super enthusiastic about. I might reduce the extent of the change to make life easier though

**Pull request merged**